src/ng/directive/ngModel.js

```diff
@@ -829,6 +829,29 @@ NgModelController.prototype = {
         that.$commitViewValue();
       });
     }
+  },
+
+  /**
+   * @ngdoc method
+   *
+   * @name ngModel.NgModelController#$overrideModelOptions
+   *
+   * @description
+   *
+   * Override the current model options settings programmatically.
+   *
+   * The previous `ModelOptions` value will not be modified. Instead, the
+   * new `ModelOptions` object will inherit from the previous value of
+   * `NgModelController.$options`.
```

Comment:
IIRC, this is the only place we publicly document `NgModelController.$options`. I don't think it is a good idea (especially considering we don't specify what its type is or what the API of this `ModelOptions` type is).